### DISCUSSION TITLE
x-focus attribute to work in conjunction with x-show

### DISCUSSION THREAD

**Discussion by earthboundkid** [alpinejs/alpine#772] [4. General]
When making a modal or anything that appears and disappears, a lot of my JS ends up just being bookkeeping around what has focus. How about this for time saver: when x-show is toggled to true, it querySelects for a child with x-focus. If it finds one, the current document.activeElement is put into a stack somewhere and that item is focused. On show==false, it restores the old document.activeElement, if it exists.

What do you think?

**Comment by SimoTod** [comment]
It's a very specific implementation that fits your use case but we can't force that behaviour for all user. You can achieve the same result using nextTick and settings the focus either when you toggle the variable or using a watcher

**Comment by earthboundkid** [comment]
Right. I'm just proposing that this is a common need and would let more people get away with components that are just `x-data="{ isOpen: false }"` instead of needing to go through the whole thing with nextTick.

**Reply by SimoTod** [reply]
Yeah, I get it. I was trying to figure out what happens for those who don't need it. You wouldn't be able to disable it while with the current implementation people can implement their own behaviour. 

It's good to open a discussion though, the message above was just my personal opinion, others may think differently.
